swift/Workflow/Sources/RunLoopObserver.swift

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CoreFoundation
+
+/// Swift wrapper for `CFRunLoopObserver`
+internal final class RunLoopObserver {
+    private var observer: CFRunLoopObserver!
```

Comment:
nit: I think this can be a `let` without the `!` since nothing in the initializer needs `self`?

Reply:
Oh, yes, it can probably be a `let`. Sorry, this went through a few revisions. The `CFRunLoopObserverCreateWithHandler` returns an implicitly unwrapped optional, so we need to handle that somewhere (maybe this should be a `let observer: CFRunLoopObserver?`).

Reply:
Yeah, that was my guess.

Looks like that function doesn’t have nullability annotations. I’d say we should make this `guard let … else fatalError` since the documentation mentions no scenario in which it will return null. And without this observer we’re going to have a very broken pile of software.

Reply:
True. It feels a bit weird to add it to this mostly-generic RAII wrapper for run loop observers, but I guess I could make the initializer failable and we could `fatalError` in `WorkflowHost`.